jakobo/expo-community-flipper
# Compatibility Changes for React Native 0.69.2+ (SDK46)

This will likely address the issues in #18 and #19.

Context: In React Native 0.69.2, the way flipper was added to the build for iOS was changed in a backward incompatible way. Previously, flipper was invoked with a `use_flipper` invocation in the Podfile. The [upgrade helper](https://react-native-community.github.io/upgrade-helper/?from=0.68.2&to=0.69.4) shows that the file changed significantly at the 0.69.2 release of react native

* Flipper is now invoked as an argument to `use_react_native` and the old `use_flipper` is removed
* The `post_install` step is no longer required

The result is inconsistent builds on iOS, where the iOS build will fail while Android succeeds. Forcing old versions of flipper has limited success, but isn't a good long term solution.

# How to Test This Change

1. Check out this PR somewhere locally
2. Copy the `/plugin/build` folder into your expo app
3. Change the plugin to match the new syntax (no need to specify a Flipper version anymore)

```json
{
  "expo": {
    "..."
    "plugins": [
      "./path/to/build/withFlipper.js"
    ]
  }
}
```

You can then run `expo prebuild` and verify the new Podfile changes.

1. A `expo-community-flipper-isprod` tag that tells `use_react_native` if the production ENV value was set
2. A `expo-community-flipper-urn` tag that tells `use_react_native` to enable/disable flipper based on the `FLIPPER_DISABLE` environment variable
